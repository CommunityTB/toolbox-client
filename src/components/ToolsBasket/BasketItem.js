import React, { Component } from 'react';
import { AppContext } from '../AppProvider';

class BasketItem extends Component {
  static contextType = AppContext

  handleClick = (e) => {
    e.preventDefault();
    console.log('Remove item id', this.props.tool.id)
    this.context.actions.removeFromBasket(this.props.tool.id)
  }

  render() {
    const { tool, checkoutStatus } = this.props
    const imageUrl = `https://res.cloudinary.com/tooltimeshare/image/upload/w_48/tools/${tool.tool_img_filename}`;
    const removeBtn = checkoutStatus ? null : <a href="/" onClick={(e)=>this.handleClick(e)}><span className="x-symbol">&#x2715;</span></a>
    return (
      <div className="tool-line-item">
        <div className="media">
          <img src={imageUrl} alt={tool.tool_img_alt} />
          <div className="title">
            <p>{tool.tool_name}</p>
          </div>
        </div>
        <div className="remove-from-basket">{removeBtn}</div>
      </div>
    );
  }
}

export default BasketItem;