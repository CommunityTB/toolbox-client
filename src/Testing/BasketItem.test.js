import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import BasketItem from '../components/ToolsBasket/BasketItem'

describe(`BasketItem component`, () => {
  const tool = {
    tool_img_filename: 'drill.jpg',
    id: '1',
    tool_img_alt: 'drill',
    tool_name: 'drill',
    return_date: null,
  }

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BasketItem tool={tool} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a BasketItem by default', () => {
    const wrapper = shallow(<BasketItem tool={tool} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})