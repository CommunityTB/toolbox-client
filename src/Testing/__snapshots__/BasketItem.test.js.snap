// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketItem component renders a BasketItem by default 1`] = `
<div
  className="tool-line-item"
>
  <div
    className="media"
  >
    <img
      alt="drill"
      src="https://res.cloudinary.com/tooltimeshare/image/upload/w_48/tools/drill.jpg"
    />
    <div
      className="title"
    >
      <p>
        drill
      </p>
    </div>
  </div>
  <div
    className="remove-from-basket"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <span
        className="x-symbol"
      >
        âœ•
      </span>
    </a>
  </div>
</div>
`;
